### ISSUE TITLE
Add submodule-inclusive option to "clean working directory"

### ISSUE THREAD

**Issue by Piedone** [gitextensions/gitextensions#8011]
## Feature description

The Commands → "Clean working directory..." feature performs a "shallow" clean, it won't clean submodules. Thus you need to run it for each submodule too. The proposal is about adding the option to this feature to also clean submodules.

The dialog could have a checkbox like "Clean submodules". Cleaning submodules would perform the exact same command for each submodule too.

Note that until this is not available you [can use a script for this](https://github.com/gitextensions/gitextensions/issues/3314#issuecomment-616170601).

## Environment

- Git Extensions 3.3.1.7897
- Build 5a97671645532bcedc443bac7b727f40db47cb5c
- Git 2.23.0.windows.1
- Microsoft Windows NT 10.0.18363.0
- .NET Framework 4.8.4150.0
- DPI 96dpi (no scaling)

**Comment by gerhardol** [comment]
Reasonable
The hard part is what to call the option?
```Recursive in submodules```

**Comment by Piedone** [comment]
Yeah, cache invalidation and naming things :).

It's fine for me, however, my nitpick would be that `--recursive` is a switch on the `git submodule foreach` command so for certain people, it might be confusing: Does it mean that submodules are also cleaned or submodules are always cleaned but this will perform a recursive clean within submodules, i.e. cleaning submodules nested within submodules too (which I think should happen in any case)?

So perhaps "Also clean submodules, recursively"? Something that makes it apparent that a) this is in addition to the root repo clean and b) it'll cover nested submodules.

**Comment by RussKie** [comment]
"Clean submodules recursively" sounds good.

Please post a mockup screenshot to visualise the proposal.

Thank you

**Comment by Piedone** [comment]
UI design is not exactly my expertise but I'd imagine something like this:
![image](https://user-images.githubusercontent.com/1976647/79871772-29c8db00-83e5-11ea-9e1b-38c7d52ce816.png)

**Comment by RussKie** [comment]
Acceptable 👍

**Comment by gerhardol** [comment]
Put the text after Remove untracked
Good

**Comment by RussKie** [comment]
While we at it, the [Cancel] button should be removed

**Comment by gerhardol** [comment]
> While we at it, the [Cancel] button should be removed

Why, we have this on most forms that may perform an action?

**Comment by RussKie** [comment]
> Why, we have this on most forms that may perform an action?

No, we don't - https://github.com/gitextensions/gitextensions/issues/6183. We have only 3 or 4 forms with it.

**Comment by mstv** [comment]
If the GUI experts say "`Cancel` buttons are obsolete", I have no concerns as long as these dialogs can be closed using `Esc`.
(I guessed GE had more such forms.)

**Comment by RussKie** [comment]
I'm not saying [Cancel] button is obsolete, but it is inconsistent with our current designs (which are all over the place).
It _may_ be good to have it but we need to have it consistently.
